sample-kotlin/src/main/java/com/fblitho/lithoktsample/lithography/data/DataFetcher.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ *
+ *  This file provided by Facebook is for non-commercial testing and evaluation
+ *  purposes only.  Facebook reserves all rights not expressly granted.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ *  FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *  ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+package com.fblitho.lithoktsample.lithography.data
+
+import android.arch.lifecycle.MutableLiveData
+import android.os.AsyncTask
+
+class DataFetcher(val model: MutableLiveData<Model>) : Fetcher {
+
+  var fetching = -1
+
+  override fun fetchMoreData(lastFetchedDecade: Int) {
+    if (fetching == lastFetchedDecade + 1) {
+      return
+    }
+
+    fetching = lastFetchedDecade + 1;
+
+    model.value = Model(model.value!!.decades, true)
+    object : AsyncTask<Void, Void, List<Decade>>() {
+      override fun doInBackground(vararg params: Void?): List<Decade> {
+        //Let's simulate a network call here.
+        Thread.sleep(2000);
+        return DataCreator.createPageOfData(lastFetchedDecade + 1);
+      }
+
+      override fun onPostExecute(result: List<Decade>?) {
+        model.value = Model(model.value!!.decades + result!!, false)
```

Comment:
You can also avoid NPE here for null `value` by writing something like: 

```kotlin
val decades = model.value?.decades ?: emptyList()
model.value = Model(decades + result!!, false)
```

Reply:
Love it!